### PR TITLE
Fix shellwords delimiter handling

### PR DESCRIPTION
**Pull request by A-Walrus**
This allows commands such as `:set statusline.center ["file-type"]` to work. Before the quotes within the list would mess it up. Also added a test to ensure correct behavior.
Addresses the parsing part of #2206 (the not updating is solved by #3457)

### CODE REVIEW

File: helix-core/src/shellwords.rs [hunk 1]
```diff
@@ -3,8 +3,9 @@ use std::borrow::Cow;
 /// Get the vec of escaped / quoted / doublequoted filenames from the input str
 pub fn shellwords(input: &str) -> Vec<Cow<'_, str>> {
     enum State {
-        Normal,
-        NormalEscaped,
+        Delimiter, // whitespace
```
Comment (the-mikedavis) [inline helix-core/src/shellwords.rs:6]: To align with the other states I think we should call this "Delimited" or maybe "OnWhitespace"

**Comment by the-mikedavis** [comment]
I think we start to diverge from the behavior of quoting within shells here, right?

```bash
$ echo ["file-type","file-encoding"]
[file-type,file-encoding]
```

Although it might be worth the tradeoff for convenience. I suppose that the alternative is that we recommend single quotes for using `:set`, for example `:set statusline.center '["file-type","file-encoding"]'`

**Comment by A-Walrus** [comment]
> I think we start to diverge from the behavior of quoting within shells here, right?

Yeah, we do. I see three options

-  Diverge from  shell behavior for convenience
- Keep current behavior and document and recommend wrapping lists in single quotes
- Special handling for lists by the actual command so that `:set statusline.center file-type file-encoding` works. It would receive them as separate args from the parser, see that the keys value is of type list, and try putting all the remaining args into the list. 

@the-mikedavis it's your call, whatever you think makes sense.

**Review by archseer — APPROVED** [review]
LGTM, I think it's ok to diverge from regular shell behaviour

**Pull request merged**